tools/python_tools/cuvslam_tools/tests/test_icl_nuim_conversion.py

```
@@ -0,0 +1,401 @@
+
+    def test_nested_layout_is_accepted(self):
+        destination = self._extracted(prefix="traj0_frei_png")
+        with unittest.mock.patch.object(icl_prepare, "extract_tar_archive"):
+            found = icl_prepare.extract_sequence(Path("archive.tar.gz"), destination)
+        self.assertEqual(found.name, "traj0_frei_png")
+
+    def test_archive_without_both_streams_is_reported(self):
+        destination = self.root / "out"
+        (destination / "rgb").mkdir(parents=True)
+        with unittest.mock.patch.object(icl_prepare, "extract_tar_archive"):
+            with self.assertRaisesRegex(Exception, "rgb/ and depth/"):
```

Ruff (0.16.3)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)